docs/tech-specs/basic-spec.md

```diff
@@ -6,6 +6,269 @@
+
+### Prioritized Message Limitations
+
+Taking priority into consideration automatically will have throughput implications since enqueueing and dequeueing will have to be mutually exclusive across the queue. And trying to maintain the queue across the cluster of broker would add more complication. So we will limit the priority to take effect within a single instance of the broker process. What it means is when we have a worker available to dispatch a message it will always give priority to the highest priority message. It also means, if there are too many messages queued, the likely time needed to queue will be longer; so increasing brokers aggressively is a deployment recommendation.
+
+### HTTP2 Consideration
+
+Given throughput is one of the primary goal of the project, HTTP/2 had to be an avenue we explore to take advantage of its multiplexing capabilities and binary frame compression. The limitation would be what if a consumer is not HTTP/2 enabled or the producer's HTTP client is HTTP/1.1 compliant only; in most cases Go's HTTP server and client can talk in both protocols. One other consideration is, the load balancer used with the broker has to be HTTP/2 compliant in receiving and proxying; for example, until [very recently](https://aws.amazon.com/blogs/aws/new-application-load-balancer-support-for-end-to-end-http-2-and-grpc/) AWS ALB is HTTP/2 enabled in receiving but not when proxying. From an implementation standpoint we will have to account for the fact that now requests [will come more in batch](https://www.lucidchart.com/techblog/2019/04/10/why-turning-on-http2-was-a-mistake/) rather than sequential calls.
+
+## Implementation Details
+
+### Objects
+
+![Class Diagram](./images/basic-spec/BrokerObjects.svg)
+
+### Endpoints
+
+We want the broker service itself to be stateless and follow the RESTful API paradigms. As follows are the conventions for the API -
+
+* `PUT` for creating **Producer**, **Channel** and **Consumer**; primarily to allow the client to dictate the ID; it will be idempotent so it can work as edit as well
+* All `PUT` request will accept `application/x-www-form-urlencoded` MIME Type Form data
+* All the `GET` will expose a `Last-Modified` header
+* `DELETE` is not supported for any resource type
```

Comment:
Shouldn't you be able to delete a consumer if you don't want it to receive messages anymore?

Reply:
This is a good point; will add it.